Mathlib/Data/Set/Pointwise/Basic.lean

```diff
@@ -1311,13 +1311,9 @@ theorem preimage_mul_preimage_subset {s t : Set β} : m ⁻¹' s * m ⁻¹' t 
 
 @[to_additive]
 lemma preimage_mul (hm : Injective m) {s t : Set β} (hs : s ⊆ range m) (ht : t ⊆ range m) :
-    m ⁻¹' (s * t) = m ⁻¹' s * m ⁻¹' t := by
-  refine subset_antisymm ?_ (preimage_mul_preimage_subset m)
-  rintro a ⟨b, c, hb, hc, ha⟩
-  obtain ⟨b, rfl⟩ := hs hb
-  obtain ⟨c, rfl⟩ := ht hc
-  simp only [← map_mul, hm.eq_iff] at ha
-  exact ⟨b, c, hb, hc, ha⟩
+    m ⁻¹' (s * t) = m ⁻¹' s * m ⁻¹' t := hm.image_injective <| by
+  rw [image_mul, image_preimage_eq_iff.2 hs, image_preimage_eq_iff.2 ht,
+    image_preimage_eq_iff.2 (mul_subset_range m hs ht)]
```

Comment:
I think I'd prefer this formatted as
```suggestion
    m ⁻¹' (s * t) = m ⁻¹' s * m ⁻¹' t :=
  hm.image_injective <| by
    rw [image_mul, image_preimage_eq_iff.2 hs, image_preimage_eq_iff.2 ht,
      image_preimage_eq_iff.2 (mul_subset_range m hs ht)]
```
as otherwise it's easy to miss what's going on in the proof.

Reply:
(and below)